framework/include/kernels/MatDiffusionBase.h

```diff
@@ -52,20 +52,27 @@ class MatDiffusionBase : public DerivativeMaterialInterface<JvarMapKernelInterfa
   const bool _is_coupled;
 
   /// int label for the Concentration
-  unsigned int _conc_var;
+  unsigned int _v_var;
 
   /// Gradient of the concentration
-  const VariableGradient & _grad_conc;
+  const VariableGradient & _grad_v;
 };
 
 template <typename T>
 InputParameters
 MatDiffusionBase<T>::validParams()
 {
   InputParameters params = ::validParams<Kernel>();
```

Comment:
What do the double colons do here?

Reply:
The resolve to the global namespace. This is needed to differentiate from the class's own validParams member function.